### PR TITLE
Add app.rb file and main.rb file

### PR DESCRIPTION
**Pull request by Haadiiii**
## OOP School library Basic API :+1: :arrow_double_down: 

In this Activity : 

> APP.rb

- Create an `app.rb` file that will serve as your console app entry-point. It have methods that do the following:
  - List all books.
  - List all people.
  - Create a person (teacher or student, not a plain `Person`).
  - Create a book.
  - Create a rental.
  - List all rentals for a given person's id.

> MAIN.rb

- In `main.rb` define the entry point, this will be a method called `main` that is invoked at the end of your file. This method do the following:
  - Present the user with a list of options to perform.
  - Lets users choose an option.
  - If needed, ask for parameters for the option.
  - Have a way to quit the app.

### CODE REVIEW

File: app.rb [hunk 1]
```diff
@@ -0,0 +1,147 @@
...
+    when 2
+      puts 'Great! let\'s create the Teacher!'
+      print 'Teacher age: '
+      teacher_age = gets.chomp.to_i
+      print 'Teacher name: '
+      teacher_name = gets.chomp
+      print 'Teacher specialization: '
+      teacher_specs = gets.chomp
+      teacher_permission = true
+      @people.push(Teacher.new(teacher_age, teacher_name, teacher_specs, 'junior', teacher_permission))
```
Comment (brytebee) [inline app.rb:85]: - [ ] **Required:** The Teacher class expects 4 inputs but you've supplied 5 here and so the app crashes in an attempt to create them. Kindly fix it.

image

File: app.rb [hunk 2]
```diff
@@ -0,0 +1,147 @@
...
+    case stdparpermission
+    when 'true'
+      @people.push(Student.new(stdage, std_class, stdname, true))
+    when 'false'
+      @people.push(Student.new(stdage, std_class, stdname, false))
+    else
+      puts 'That was an invalid entry'
+    end
+
+    puts 'Student is created successfully'
```
Comment (brytebee) [inline app.rb:62]: - [ ] **Required:** When a Student isn't created successfully, the app still prints these lines. This is very confusing for users. Kindly fix it.

image

File: app.rb [hunk 3]
```diff
@@ -0,0 +1,147 @@
...
+    if @people.length.zero?
+      puts 'You need at least one person'
+    else
+      @people.map.each_with_index do |person, index|
+        puts "#{index}) [#{person.classroom}] Name: #{person.name}, ID: #{person.id}, Age: #{person.age}"
+      end
+    end
+  end
+
+  def create_student
```
Comment (brytebee) [inline app.rb:43]: - [ ] **Required:** I cannot create a student successfully, kindly look long into it and fix it.

image

**Review by brytebee — CHANGES_REQUESTED** [review]
Hi, @Haadiiii👋,

##

### Highlights💡
- Good commit message.✔️ 
- Good PR Summary.✔️
- The Readme looks professional.✔️
- Great UI design.✔️
- Book creation is successful.✔️

##

Good job so far coming so far with the progress you've made! 💪💪
 
However, there are some issues that you still need to work on to go to the next project but you are almost there!



# Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

Please, remember to tag me in your question so I can receive the notification.

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me @brytebee in your question so I can receive the notification.
### _**Use @brytebee on slack for faster responses.**_


**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: app.rb (hunk 1 above)
Comment (Haadiiii) [reply app.rb]: got it :+1:

File: app.rb (hunk 3 above)
Comment (Haadiiii) [reply app.rb:43]: resolved :+1:

File: app.rb (hunk 2 above)
Comment (Haadiiii) [reply app.rb]: solved :+1:

**Pull request merged**